models/business-unit/src/associate-role-assignment/transformers.ts

```diff
@@ -0,0 +1,53 @@
+import { type AssociateRoleKeyReference } from '@commercetools/platform-sdk';
+import { KeyReference } from '@commercetools-test-data/commons';
+import { Transformer } from '@commercetools-test-data/core';
+import type {
+  TAssociateRoleAssignmentDefault,
+  TAssociateRoleAssignmentRest,
+  TAssociateRoleAssignmentGraphql,
+} from './types';
+
+const transformers = {
+  default: Transformer<
+    TAssociateRoleAssignmentDefault,
+    TAssociateRoleAssignmentDefault
+  >('default', {
+    buildFields: ['associateRole'],
+  }),
+  rest: Transformer<
+    TAssociateRoleAssignmentDefault,
+    TAssociateRoleAssignmentRest
+  >('rest', {
+    buildFields: ['associateRole'],
+    replaceFields: ({ fields }) => {
+      const associateRole: AssociateRoleKeyReference = KeyReference.presets
+        .associateRole()
+        .key(fields.associateRole.key)
+        .build();
```

Comment:
It might be safer to use the `id` based reference here because `key` is an optional field.

Reply:
To clarify do you mean that `fields.associateRole.key` is optional, or that key is optional in `AssociateRoleKeyReference`?

Reply:
Sorry to highjack, in [`AssociateRoles`](https://docs.commercetools.com/api/projects/associate-roles#associaterole) the `key` is mandatory